android/src/main/java/com/airbnb/android/react/maps/AirMapView.java

```diff
@@ -289,6 +296,24 @@ private boolean hasPermissions() {
                 checkSelfPermission(getContext(), PERMISSIONS[1]) == PackageManager.PERMISSION_GRANTED;
     }
 
+    /*
+    onDestroy is final method so I can't override it.
+     */
+    public synchronized  void doDestroy() {
+        if (lifecycleListener != null && context != null) {
+            context.removeLifecycleEventListener(lifecycleListener);
+            lifecycleListener = null;
+        }
+        if(!paused) {
+            onPause();
+        }
+        if (!destroyed) {
+            onDestroy();
+            destroyed = true;
```

Comment:
You can just set destroyed to true at the top of the method. This will prevent future logic errors if you ever return early or anything